dspace-api/src/main/java/org/dspace/content/RelationshipMetadataServiceImpl.java

```diff
@@ -0,0 +1,285 @@
+                                                                         VirtualMetadataConfiguration virtualBean,
+                                                                         String value) {
+        RelationshipMetadataValue metadataValue = constructMetadataValue(context, key);
+        if (metadataValue != null) {
+            metadataValue = constructResultingMetadataValue(item, value, metadataValue, relationshipId);
+            metadataValue.setUseForPlace(virtualBean.getUseForPlace());
+            metadataValue.setPlace(place);
+            if (StringUtils.isNotBlank(metadataValue.getValue())) {
+                return metadataValue;
+            }
+        }
+        return null;
+    }
+
+    //This method will construct a RelationshipMetadataValue object with proper schema, element and qualifier based
+    //on the key String parameter passed along to it
+    private RelationshipMetadataValue constructMetadataValue(Context context, String key) {
+        String[] splittedKey = key.split("\\.");
+        RelationshipMetadataValue metadataValue = new RelationshipMetadataValue();
+        String metadataSchema = splittedKey.length > 0 ? splittedKey[0] : null;
+        String metadataElement = splittedKey.length > 1 ? splittedKey[1] : null;
+        String metadataQualifier = splittedKey.length > 2 ? splittedKey[2] : null;
+        MetadataField metadataField = null;
+        try {
+            metadataField = metadataFieldService
+                .findByElement(context, metadataSchema, metadataElement, metadataQualifier);
+        } catch (SQLException e) {
+            log.error("Could not find element with MetadataSchema: " + metadataSchema +
+                ", MetadataElement: " + metadataElement + " and MetadataQualifier: " + metadataQualifier, e);
+            return null;
+        }
+        if (metadataField == null) {
+            log.error("A MetadataValue was attempted to construct with MetadataField for parameters: " +
+                    "metadataschema: {}, metadataelement: {}, metadataqualifier: {}",
+                metadataSchema, metadataElement, metadataQualifier);
+            return null;
+        }
+        metadataValue.setMetadataField(metadataField);
+        metadataValue.setLanguage(Item.ANY);
+        return metadataValue;
+    }
+
+
+    private RelationshipMetadataValue constructResultingMetadataValue(Item item, String value,
+                                                                      RelationshipMetadataValue metadataValue,
+                                                                      Integer relationshipId) {
+        metadataValue.setValue(value);
+        metadataValue.setAuthority(Constants.VIRTUAL_AUTHORITY_PREFIX + relationshipId);
+        metadataValue.setConfidence(-1);
+        metadataValue.setDSpaceObject(item);
+        return metadataValue;
+    }
+
+
+    private RelationshipMetadataValue getRelationMetadataFromOtherItem(Context context, Item otherItem,
```

Comment:
Could you please add javadocs?

Reply:
This is a private methods, but a short explanation has been included

Reply:
This is a private method, but some comments have been included now as well